src/revpi-factory-reset.sh

```diff
@@ -41,7 +52,7 @@ while [ ! -r /home/pi/.revpi-factory-reset ] ; do
 
 	# this creates /home/pi/.revpi-factory-reset on success:
 	/usr/bin/sudo /usr/sbin/revpi-factory-reset "$ovl" "$ser" "$mac" 2>/dev/null
-	if [ "$?" = 1 ]; then
+	if [ "$?" == "1" ]; then
```

Comment:
Please put this change in a separate commit!

Reply:
You can squash this change into my commit (https://github.com/RevolutionPi/piserial/pull/47/commits/f64e57abbc99babd54ada70d330203c63fe73dbb)  if you like

Reply:
No, I won't sqash since this fixes an error I so far only had after my initial commit with CM1 only. This is why I re-ordered the commits after separation.

Reply:
Is there any special reason for treating this as string and not something like this?

```bash
if [ $? -eq 1 ]; then 
```

Reply:
Not any I can think of. This script obviously has the typical problems of people working too long with systems where /bin/sh points to /bin/bash. As you can see in my commit message of [e91a223](https://github.com/RevolutionPi/piserial/pull/47/commits/e91a22364c589c14f97a34b361a5991e060dccb7), I only changed this for consistency. I otherwise would have rewritten huge parts of the script.

Reply:
I would be OK with your solution. Just wondering  :smile: